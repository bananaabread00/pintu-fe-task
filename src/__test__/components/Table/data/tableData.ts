import { HeadCell } from '../../../../interfaces/table';

interface DessertList {
  name: string;
  calories: number;
  fat: number;
}

export const headCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert'
  },
  {
    id: 'calories',
    label: 'Calories',
    sortable: true
  },
  {
    id: 'fat',
    label: 'Fat (g)',
    sortable: true,
    width: 250
  }
];

export const tableData = [
  {
    name: 'Frozen yoghurt',
    calories: 100,
    fat: 80
  },
  {
    name: 'Ice cream',
    calories: 180,
    fat: 97
  },
  {
    name: 'Eclair',
    calories: 235,
    fat: 98
  }
];
