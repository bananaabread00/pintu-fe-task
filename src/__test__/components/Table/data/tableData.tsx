import React from 'react';
import { HeadCell } from '../../../../interfaces/table';

interface DessertList {
  name: string;
  calories: number;
  fat: number;
}

export const sortableHeadCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert',
    render: (value) => <strong>{value}</strong>
  },
  {
    id: 'calories',
    label: 'Calories',
    sortable: true
  },
  {
    id: 'fat',
    label: 'Fat (g)',
    sortable: true,
    width: 250
  }
];

export const dropdownHeadCells: HeadCell<DessertList>[] = [
  {
    id: 'name',
    label: 'Dessert',
    main: true
  },
  {
    id: 'calories',
    label: 'Calories'
  }
];

export const tableData = [
  {
    name: 'Frozen yoghurt',
    calories: 100,
    fat: 80
  },
  {
    name: 'Ice cream',
    calories: 180,
    fat: 97
  },
  {
    name: 'Eclair',
    calories: 235,
    fat: 98
  }
];

export const dropdownList = [
  { key: 'calories', label: 'Calories' },
  { key: 'fat', label: 'Fat (g)' }
];
