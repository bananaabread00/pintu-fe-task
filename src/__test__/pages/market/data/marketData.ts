export const getSupportedCurrenciesData = {
  code: 'success',
  payload: [
    {
      currencyGroup: 'BTC',
      color: '#F78B1A',
      currencySymbol: 'BTC',
      name: 'Bitcoin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BTC.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:45Z',
      wallets: [
        {
          currencyGroup: 'BTC',
          tokenSymbol: 'BTC',
          decimal_point: 8,
          tokenType: 'Bitcoin',
          blockchain: 'Bitcoin',
          explorer: 'https://explorer.bitcoin.com/btc/tx/',
          listingDate: '2020-09-15T09:43:45Z',
          blockchainName: 'Bitcoin',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/Bitcoin.svg'
        }
      ]
    },
    {
      currencyGroup: 'ETH',
      color: '#9011FE',
      currencySymbol: 'ETH',
      name: 'Ethereum',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_ETH.svg',
      decimal_point: 18,
      listingDate: '2020-09-15T09:43:46Z',
      wallets: [
        {
          currencyGroup: 'ETH',
          tokenSymbol: 'ETH',
          decimal_point: 18,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:46Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'USDT',
      color: '#15A57C',
      currencySymbol: 'USDT',
      name: 'Tether',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_USDT.svg',
      decimal_point: 6,
      listingDate: '2020-09-15T09:43:47Z',
      wallets: [
        {
          currencyGroup: 'USDT',
          tokenSymbol: 'USDT',
          decimal_point: 6,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:47Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'BNB',
      color: '#FEBF11',
      currencySymbol: 'BNB',
      name: 'Binance Coin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BNB.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:48Z',
      wallets: [
        {
          currencyGroup: 'BNB',
          tokenSymbol: 'BNB',
          decimal_point: 8,
          tokenType: 'BEP-2',
          blockchain: 'Binance',
          explorer: 'https://explorer.binance.org/tx/',
          listingDate: '2020-09-15T09:43:48Z',
          blockchainName: 'BNB Beacon Chain',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/BEP-2.svg'
        }
      ]
    }
  ],
  message: ''
};

export const getPriceChangesData = {
  code: 'success',
  payload: [
    {
      pair: 'btc/idr',
      latestPrice: '10268',
      day: '-4.84',
      week: '-7.39',
      month: '-7.64',
      year: '-24.52'
    },
    {
      pair: 'eth/idr',
      latestPrice: '21904',
      day: '-0.89',
      week: '-8.15',
      month: '-5.24',
      year: '-54.20'
    },
    {
      pair: 'usdt/idr',
      latestPrice: '28825405',
      day: '0.09',
      week: '1.32',
      month: '0.95',
      year: '11.00'
    },
    {
      pair: 'bnb/idr',
      latestPrice: '104711228',
      day: '-3.15',
      week: '-4.57',
      month: '18.85',
      year: '7.83'
    }
  ],
  message: ''
};
