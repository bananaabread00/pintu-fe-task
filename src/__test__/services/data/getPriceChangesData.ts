export const getPriceChangesData = {
  code: 'success',
  payload: [
    {
      pair: 'btc/idr',
      latestPrice: '10268',
      day: '-4.84',
      week: '-7.39',
      month: '-7.64',
      year: '-24.52'
    },
    {
      pair: 'eth/idr',
      latestPrice: '21904',
      day: '-0.89',
      week: '-8.15',
      month: '-5.24',
      year: '-54.20'
    },
    {
      pair: 'usdt/idr',
      latestPrice: '28825405',
      day: '0.09',
      week: '1.32',
      month: '0.95',
      year: '11.00'
    },
    {
      pair: 'bnb/idr',
      latestPrice: '104711228',
      day: '-3.15',
      week: '-4.57',
      month: '18.85',
      year: '7.83'
    }
  ],
  message: ''
};
