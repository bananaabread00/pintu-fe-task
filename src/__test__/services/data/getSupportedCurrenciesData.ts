export const getSupportedCurrenciesData = {
  code: 'success',
  payload: [
    {
      currencyGroup: 'BTC',
      color: '#F78B1A',
      currencySymbol: 'BTC',
      name: 'Bitcoin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BTC.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:45Z',
      wallets: [
        {
          currencyGroup: 'BTC',
          tokenSymbol: 'BTC',
          decimal_point: 8,
          tokenType: 'Bitcoin',
          blockchain: 'Bitcoin',
          explorer: 'https://explorer.bitcoin.com/btc/tx/',
          listingDate: '2020-09-15T09:43:45Z',
          blockchainName: 'Bitcoin',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/Bitcoin.svg'
        }
      ]
    },
    {
      currencyGroup: 'ETH',
      color: '#9011FE',
      currencySymbol: 'ETH',
      name: 'Ethereum',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_ETH.svg',
      decimal_point: 18,
      listingDate: '2020-09-15T09:43:46Z',
      wallets: [
        {
          currencyGroup: 'ETH',
          tokenSymbol: 'ETH',
          decimal_point: 18,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:46Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'USDT',
      color: '#15A57C',
      currencySymbol: 'USDT',
      name: 'Tether',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_USDT.svg',
      decimal_point: 6,
      listingDate: '2020-09-15T09:43:47Z',
      wallets: [
        {
          currencyGroup: 'USDT',
          tokenSymbol: 'USDT',
          decimal_point: 6,
          tokenType: 'ERC-20',
          blockchain: 'Ethereum',
          explorer: 'https://etherscan.io/tx/',
          listingDate: '2020-09-15T09:43:47Z',
          blockchainName: 'Ethereum',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/ERC-20.svg'
        }
      ]
    },
    {
      currencyGroup: 'BNB',
      color: '#FEBF11',
      currencySymbol: 'BNB',
      name: 'Binance Coin',
      logo: 'https://s3-ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/circle_BNB.svg',
      decimal_point: 8,
      listingDate: '2020-09-15T09:43:48Z',
      wallets: [
        {
          currencyGroup: 'BNB',
          tokenSymbol: 'BNB',
          decimal_point: 8,
          tokenType: 'BEP-2',
          blockchain: 'Binance',
          explorer: 'https://explorer.binance.org/tx/',
          listingDate: '2020-09-15T09:43:48Z',
          blockchainName: 'BNB Beacon Chain',
          logo: 'https://s3.ap-southeast-1.amazonaws.com/static.pintu.co.id/assets/images/logo/blockchain/BEP-2.svg'
        }
      ]
    }
  ],
  message: ''
};
