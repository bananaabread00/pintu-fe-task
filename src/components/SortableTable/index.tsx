import * as React from 'react';
import Box from '@mui/material/Box';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableSortLabel from '@mui/material/TableSortLabel';
import Paper from '@mui/material/Paper';
import { visuallyHidden } from '@mui/utils';
import { EnhancedTableProps, Order, SortableTableProps } from '../../interfaces/sortableTable';

function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function getComparator<Key extends keyof any>(
  order: Order,
  orderBy: Key
): (a: { [key in Key]: unknown }, b: { [key in Key]: unknown }) => number {
  return order === 'desc'
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  const stabilizedThis = array.map((el, index) => [el, index] as [T, number]);
  stabilizedThis.sort((a, b) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) {
      return order;
    }
    return a[1] - b[1];
  });
  return stabilizedThis.map((el) => el[0]) as T[];
}

function EnhancedTableHead<T>(props: EnhancedTableProps<T>) {
  const { order, orderBy, onRequestSort, headCells } = props;
  const createSortHandler = (property: keyof T) => (event: React.MouseEvent<unknown>) => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map((headCell) => (
          <TableCell
            key={headCell.id as string}
            sx={{ py: 4 }}
            sortDirection={orderBy === headCell.id ? order : false}
          >
            {headCell.sortable ? (
              <TableSortLabel
                active={orderBy === headCell.id}
                direction={orderBy === headCell.id ? order : 'asc'}
                onClick={createSortHandler(headCell.id)}
              >
                {headCell.label}
                {orderBy === headCell.id ? (
                  <Box component="span" sx={visuallyHidden}>
                    {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                  </Box>
                ) : null}
              </TableSortLabel>
            ) : (
              <span>{headCell.label}</span>
            )}
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

function SortableTable<T>(props: SortableTableProps<T>) {
  const { headCells, tableData } = props;

  const [order, setOrder] = React.useState<Order>('asc');
  const [orderBy, setOrderBy] = React.useState<keyof T | undefined>();

  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof T) => {
    const isAsc = orderBy === property && order === 'asc';
    setOrder(isAsc ? 'desc' : 'asc');
    setOrderBy(property);
  };

  const defaultRender = (value: any) => <span>{value}</span>;

  return (
    <Box sx={{ width: '100%' }}>
      <Paper sx={{ width: '100%', mb: 2 }}>
        <TableContainer>
          <Table sx={{ minWidth: 750 }} aria-labelledby="tableTitle" size="medium">
            <EnhancedTableHead
              order={order}
              orderBy={orderBy}
              onRequestSort={handleRequestSort}
              headCells={headCells}
            />
            <TableBody>
              {orderBy
                ? stableSort(tableData, getComparator(order, orderBy)).map((row, index) => (
                    <TableRow key={index}>
                      {headCells.map((column) => (
                        <TableCell
                          key={column.id as string}
                          sx={{ py: 3 }}
                          align={column.align || 'left'}
                        >
                          {column.render ? column.render(row) : defaultRender(row[column.id])}
                        </TableCell>
                      ))}
                    </TableRow>
                  ))
                : tableData.map((row, index) => (
                    <TableRow key={index}>
                      {headCells.map((column) => (
                        <TableCell
                          key={column.id as string}
                          sx={{ py: 3 }}
                          align={column.align || 'left'}
                        >
                          {column.render ? column.render(row) : defaultRender(row[column.id])}
                        </TableCell>
                      ))}
                    </TableRow>
                  ))}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </Box>
  );
}

export default SortableTable;
