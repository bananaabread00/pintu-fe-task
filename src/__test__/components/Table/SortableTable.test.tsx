import React from 'react';
import { render, screen } from '@testing-library/react';
import SortableTable from '../../../components/Table/SortableTable';
import { sortableHeadCells, tableData } from './data/tableData';
import userEvent from '@testing-library/user-event';

describe('SortableTable component test', () => {
  it('should render the table', async () => {
    const { container } = render(
      <SortableTable headCells={sortableHeadCells} tableData={tableData} />
    );
    expect(await screen.findByText('Fat (g)')).toBeInTheDocument();
    expect(container.getElementsByClassName('MuiTableRow-root').length).toBe(4);
    expect(container).toMatchSnapshot();
  });
  it('should sort the table data', async () => {
    const { container, getAllByRole } = render(
      <SortableTable headCells={sortableHeadCells} tableData={tableData} />
    );
    const [sortButton] = getAllByRole('button');
    await userEvent.click(sortButton);
    await userEvent.click(sortButton);
    expect(container.getElementsByClassName('MuiTableRow-root').length).toBe(4);
    expect(container).toMatchSnapshot();
  });
});
