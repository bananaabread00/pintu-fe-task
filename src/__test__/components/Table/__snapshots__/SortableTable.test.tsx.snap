// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableTable component test should render the table 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-vq2emh-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root MuiTable-stickyHeader css-h6w51p-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="text-gray-600 text-center"
                >
                  Dessert
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="text-gray-600 text-center"
                  >
                    Calories
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="text-gray-600 text-center"
                  >
                    Fat (g)
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Frozen yoghurt
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  100
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  80
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Ice cream
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  180
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  97
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Eclair
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  235
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  98
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`SortableTable component test should sort the table data 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-vq2emh-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root MuiTable-stickyHeader css-h6w51p-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="text-gray-600 text-center"
                >
                  Dessert
                </span>
              </th>
              <th
                aria-sort="descending"
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="text-gray-600 text-center"
                  >
                    Calories
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc css-3l415a-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-10sdfzr-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="text-gray-600 text-center"
                  >
                    Fat (g)
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Eclair
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  235
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  98
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Ice cream
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  180
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  97
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <strong>
                  Frozen yoghurt
                </strong>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-inzptt-MuiTableCell-root"
              >
                <span>
                  100
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-cfxck-MuiTableCell-root"
              >
                <span>
                  80
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
